### DISCUSSION TITLE
Can I have both Friendly Names and the GUID?

### DISCUSSION THREAD

**Discussion by jdenicola-pa** [rappen/FetchXMLBuilder#802] [Q&A]
I am exporting Contacts for example. I want the Result View to have both the GUID of the Account (parentcustomerid) and also the "friendly name" of two custom lookup fields on my Contact entity.

If I check Appearance -> Friendly Names, it does the lookups for me (which I want for the two custom lookup fields) but then it includes the Account Name. It's OK to have the Account Name, but I ALSO need the Account ID (GUID). Is there a way to have both?

**Comment by rappen** [comment]
Hi - interesting request!
No, it is now impossible. The Friendly/Raw design you use is done down in the control ([XRMDataGridView](https://github.com/rappen/Rappen.XTB.Helper/wiki/Controls#xrmdatagridview) in [Rappen.Xtb.Helpers](https://github.com/rappen/Rappen.XTB.Helper)) so we can't use some Friendly and some Raw.
Even harder - you want to get one attribute (`parentcustomerid`) in _**two**_ columns...
Well, now impossible, but I might think about this for a while!
I'll get back if I find some ideas!

**Comment by rappen** [comment]
It's possible to do today, by this query:

```
<fetch>
  <entity name='contact'>
    <attribute name='fullname' />
    <attribute name='parentcustomerid' />
    <link-entity name='account' from='accountid' to='parentcustomerid' link-type='outer' alias='A'>
      <attribute name='accountid' />
    </link-entity>
  </entity>
</fetch>
```

Example:

![image](https://user-images.githubusercontent.com/2572253/235905600-2f07263c-3b6c-4d37-9994-3e78ce571ae6.png)
